integration-cli/docker_api_containers_test.go

```diff
@@ -817,3 +817,45 @@ func TestContainerApiPostCreateNull(t *testing.T) {
 
 	logDone("containers REST API - Create Null")
 }
+
+func TestCreateWithTooLowMemoryLimit(t *testing.T) {
+	defer deleteAllContainers()
+	config := `{
+		"Image":     "busybox",
+		"Cmd":       "ls",
+		"OpenStdin": true,
+		"CpuShares": 100,
+		"Memory":    524287
+	}`
+
+	_, body, err := sockRequestRaw("POST", "/containers/create", strings.NewReader(config), "application/json")
+	b, err2 := readBody(body)
+	if err2 != nil {
+		t.Fatal(err2)
+	}
+
+	if err == nil || !strings.Contains(string(b), "Minimum memory limit allowed is 4MB") {
+		t.Errorf("Memory limit is smaller than the allowed limit. Container creation should've failed!")
+	}
+
+	logDone("container REST API - create can't set too low memory limit")
+}
+
+func TestStartWithTooLowMemoryLimit(t *testing.T) {
```

Comment:
This isn't really what I had in mind.
`docker run` and `docker create` are the same thing (`docker run` calls create and then start).

The API allows people to pass in HostConfig on start, but the CLI does not do this.
Since memory is in HostConfig now (it didn't used to be). A test for this would be to post a HostConfig with a memory to low on start (with a container that's already been created)... does that make sense?

Let me know if you need any help with this.